frappe/integrations/doctype/google_calendar/google_calendar.py

```diff
@@ -182,204 +185,114 @@ def sync(g_calendar=None):
-				"summary": account.calendar_name,
-				"timeZone": frappe.db.get_single_value("System Settings", "time_zone"),
-			}
-			created_calendar = google_calendar.calendars().insert(body=calendar).execute()
-			frappe.db.set_value(
-				"Google Calendar", account.name, "google_calendar_id", created_calendar.get("id")
-			)
-			frappe.db.commit()
-	except HttpError as err:
-		frappe.throw(
-			_("Google Calendar - Could not create Calendar for {0}, error code {1}.").format(
-				account.name, err.resp.status
-			)
-		)
-
-
 def sync_events_from_google_calendar(g_calendar, method=None):
 	"""
 	Syncs Events from Google Calendar in Framework Calendar.
 	Google Calendar returns nextSyncToken when all the events in Google Calendar are fetched.
 	nextSyncToken is returned at the very last page
 	https://developers.google.com/calendar/v3/sync
 	"""
-	google_calendar, account = get_google_calendar_object(g_calendar)
+	account = frappe.get_doc("Google Calendar", g_calendar)
 
-	if not account.pull_from_google_calendar:
+	if job_already_enqueued("calendar_import|{}".format(account.get("name"))):
 		return
 
-	sync_token = account.get_password(fieldname="next_sync_token", raise_exception=False) or None
-	events = frappe._dict()
-	results = []
-	while True:
-		try:
-			# API Response listed at EOF
-			events = (
-				google_calendar.events()
-				.list(
-					calendarId=account.google_calendar_id,
-					maxResults=2000,
-					pageToken=events.get("nextPageToken"),
-					singleEvents=False,
-					showDeleted=True,
-					syncToken=sync_token,
-				)
-				.execute()
-			)
-		except HttpError as err:
-			msg = _("Google Calendar - Could not fetch event from Google Calendar, error code {0}.").format(
-				err.resp.status
-			)
-
-			if err.resp.status == 410:
-				set_encrypted_password("Google Calendar", account.name, "", "next_sync_token")
-				frappe.db.commit()
-				msg += " " + _("Sync token was invalid and has been resetted, Retry syncing.")
-				frappe.msgprint(msg, title="Invalid Sync Token", indicator="blue")
+	caldav_url = "https://apidata.googleusercontent.com/caldav/v2/{}/events?access_token={}".format(
+		account.user, account.get_access_token()
+	)
+	response = requests.get(
+		url=caldav_url,
+		headers={
+			"User-Agent": "Mozilla/5.0",
+			"Content-Type": "text/xml",
+			"Accept": "text/xml, text/calendar",
+		},
+	)
+	cal = Calendar.from_ical(response.content)
+	return enqueue_job(parse_calendar_events, calendar=cal, account=account)
+
+
+def parse_calendar_events(calendar, account):
+	event_list = []
+	insert_count = 0
+	for event in calendar.walk("vevent"):
+		recurrence = None
+		event_id = get_event_id(event)
+		if event.get("RRULE"):
+			recurrence = event["RRULE"]
+		if event["STATUS"] == "CONFIRMED":
+			attendees = get_event_attendees(event)
+			event_list.append(event_id)
+			if not frappe.db.exists("Event", {"google_calendar_event_id": event_id}):
+				insert_count += 1
+				insert_event_to_calendar(account, event, attendees, recurrence)
```

Comment:
1. Can be simplified.
2. `insert_event_to_calendar` and `update_event_in_calendar` are a bit confusing.